utils/compilation_filter.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-present, Facebook, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import sqlite3
+import os
+from typing import (
+    List,
+    Dict,
+)
+
+# A list of all filtered transformations.
+TRANS_LIST: List['Transformation'] = []
+
+# Mapping between added nodes and the transformation that adds these nodes.
+NODES_ADDING_MAP: Dict[str, 'Transformation'] = {}
+
+
+class Transformation:
+    """A class that represents the nodes transformation, e.g. lower,fold etc.
+
+    Public attributes:
+        addedNodes_: List[str]. Added(created) nodes of this transformation.
+        removedNodes_: List[str]. Removed(replaced) nodes of this transformation.
+        ancestors_: List['Transformation']. The ancestor transformation of current transformation.
+        scopeName_: str. The scope of current transformation.
+        transID_: str. The internal transformation ID in the database.
+        isDirectTrans_ :bool. Whether this transformation directly created/replaced the given nodeName that is passed to this script file.
+    """
+
+    def __init__(self, transID: str):
+        self.addedNodes_: List[str] = []
+        self.removedNodes_: List[str] = []
+        self.ancestors_: List['Transformation'] = []
+        self.scopeName_: str = ''
+        self.transID_: str = transID
+        self.isDirectTrans_: bool = False
+
+    def appendAddedNode(self, nodeName: str) -> None:
+        """Append the added nodes of this transformation."""
+
+        self.addedNodes_.append(nodeName)
+
+    def appendRemovedNode(self, nodeName: str) -> None:
+        """Append the removed nodes of this transformation."""
+
+        self.removedNodes_.append(nodeName)
+
+    def addAnc(self, anc: 'Transformation') -> None:
+        """Add ancestors of this transformation."""
+
+        self.ancestors_.append(anc)
+
+
+class DottyPrinter:
```

Comment:
The other tool has a `DottyPrinter` class as well. Are there any opportunities to share or reuse code? After a cursory glance, `dumpGraph` looks similar between the two classes.

Reply:
will submit a separate following pr to make the changes